sympy/core/mul.py

```diff
@@ -1351,7 +1351,7 @@ def _eval_is_integer(self):
                 b, e = a.as_base_exp()
                 if not b.is_integer or not e.is_integer: return
                 if e.is_negative:
-                    denominators.append(b)
+                    denominators.append(Pow(b, S.NegativeOne*e))
```

Comment:
You can probably safely make this `2 if a is S.Half else Pow(a, -1, evaluate=False)`.

Reply:
Fixed that and it seems to work for the test cases at least.

Reply:
Nah, the tests didn't pass... `is_odd` and `is_even` returns `None` for unevaluated results.

`Pow(a, S.NegativeOne)` is about seven times faster compared to my previous approach though.